de.fhdo.lemma.service.open_api/src/de/fhdo/lemma/service/open_api/OpenApiUtil.xtend

```diff
@@ -0,0 +1,88 @@
+package de.fhdo.lemma.service.open_api
+
+import de.fhdo.lemma.data.datadsl.extractor.DataDslExtractor
+import de.fhdo.lemma.data.DataModel
+import org.eclipse.emf.ecore.EObject
+import java.nio.file.Files
+import java.nio.file.Paths
+import de.fhdo.lemma.service.ServiceModel
+import de.fhdo.lemma.servicedsl.extractor.ServiceDslExtractor
+import de.fhdo.lemma.technology.Technology
+import de.fhdo.lemma.technology.technologydsl.extractor.TechnologyDslExtractor
+import de.fhdo.lemma.data.DataFactory
+import de.fhdo.lemma.data.PrimitiveType
+
+/**
+ * This class collects _static_ utility methods for the OpenAPI plugin.
+ *
+ * @author <a href="mailto:jonas.sorgalla@fh-dortmund.de">Jonas Sorgalla</a>
+ */
+final class OpenApiUtil {
+    /** Factory to actually create and manipulate a LEMMA DataModel */
+    static val DATA_FACTORY = DataFactory.eINSTANCE
+    static def String removeInvalidCharsFromName(String str) {
+        var ret = str;
+        ret = ret.replaceAll("[^a-zA-Z0-9_]", "");
+        if(!Character.isAlphabetic(ret.charAt(0)))
+            ret = "v"+ret;
+         return ret
+    }
+
+    /**
+     * Serialize a LEMMA data or mapping model to the given file path. Returns true if the file path
+     * and the contents extracted from the model's root are not empty.
+     */
+    static def boolean writeModel(EObject modelRoot, String filepath) {
+        if (filepath.nullOrEmpty)
+            return false
+        //if modelRoot is a DataModel, serialize it as a string
```

Comment:
This comment does not add any further information in this context and you want to remove it